src/components/stack/BlockItem.js

```diff
@@ -126,38 +134,58 @@ class BlockItem extends Component {
   }
 
   handleBurstsUpdate(newBurstsValue) {
+    const {blocks, blockId} = this.props;
     this.props.dataBlockUpdate({
+      ...blocks[blockId],
       numBursts: newBurstsValue,
     }, this.props.blockId);
     this.setState({numBursts: newBurstsValue});
   }
 
-  handleSwapBlocks(id, above) {
+  handleOrderUpdate(newOrder) {
     const {stacks, stackFocused} = this.props;
+    this.props.dataStackUpdate({
+      ...stacks[stackFocused],
+      order: newOrder,
+    }, this.props.stackFocused);
+  }
 
-    const blockIdsArray = stacks[stackFocused].order;
-    const index = blockIdsArray.indexOf(id);
-    if (index === 0 && above === true ||
-      index === blockIdsArray.length - 1 && above === false) {
-      return;
+  handleBlockDelete() {
+    const {stackFocused} = this.props;
+    this.props.dataBlockDelete({
+
+    }, this.props.blockId, stackFocused);
+  }
+
+  handleSwapBlocks(currIndex, swapIndex) {
+    const {stacks, stackFocused} = this.props;
+
+    const blocksOrderNew = [];
+    const blocksOrderOld = stacks[stackFocused].order;
+
+    // perform a deep copy
+    for (let index = 0; index < blocksOrderOld.length; index++) {
+      blocksOrderNew.push(blocksOrderOld[index]);
     }
```

Comment:
you can just use the spread operator

blocksOrderNew = [...stacks[stackFocused].order];

Reply:
Ah okay makes sense